craft_application/models/project.py

```diff
@@ -38,6 +39,12 @@
     VersionStr,
 )
 
+# The Ubuntu base that the 'devel' alias currently points to
+# TODO: This should point to 24.10, which is not available yet
+CURRENT_DEVEL_BASE = craft_providers.bases.ubuntu.BuilddBaseAlias.DEVEL
```

Comment:
Can we make this something more generic that applications could use declaratively? Ideally we'd be able to register additional unstable bases and their build bases without having to override `_validate_devel`.

Example use case: charmcraft currently supports centos 7 and almalinux 9, but if we added Alma 10 support we'd want to call it a devel base.

Reply:
What do you think about changing these to lists: `CURRENT_DEVEL_BASES` and `DEVEL_BASES`?

Alma, centos, and any other craft-providers bases can always be registered here.

When we add another devel base, we will have to validate that there is no cross-base usage like `base: almalinux@9, build-base: ubuntu@devel`.  We can check that both bases are the same craft-providers class.

Reply:
What about a single dict. Something like:

```
DEVEL_BASES = {
    BuilddBaseAlias.DEVEL: BuilddBaseAlias.DEVEL
}
```

Reply:
Even better!

Reply:
I added a data structure.  I don't love the naming but I couldn't think of anything better.